examples/gno.land/p/avl/mut_tree.gno

```diff
@@ -0,0 +1,52 @@
+package avl
+
+// MutTree
+
+type MutTree struct {
+	tree *Tree
+}
+
+func NewMutTree() *MutTree {
+	return &MutTree{
+		tree: nil,
+	}
+}
+
+func (mtree *MutTree) Size() int {
+	return mtree.tree.Size()
+}
+
+func (mtree *MutTree) Has(key string) (has bool) {
+	return mtree.tree.Has(key)
+}
+
+func (mtree *MutTree) Get(key string) (value interface{}, exists bool) {
+	_, value, exists = mtree.tree.Get(key)
+	return
+}
+
+func (mtree *MutTree) GetByIndex(index int) (key string, value interface{}) {
+	return mtree.tree.GetByIndex(index)
+}
+
+func (mtree *MutTree) Set(key string, value interface{}) (updated bool) {
+	newtree, updated := mtree.tree.Set(key, value)
+	mtree.tree = newtree
+	return updated
+}
+
```

Comment:
This will be one of the tricky cases to be supported in contract migrations; I’ll try to write tests around this.

Reply:
how so?